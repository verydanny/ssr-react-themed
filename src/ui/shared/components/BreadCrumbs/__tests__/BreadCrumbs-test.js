import React from 'react'
import { shallow, mount } from 'enzyme'
import ThemedBreadCrumbs from '../BreadCrumbs'

const BreadCrumbs = ThemedBreadCrumbs.WrappedComponent

describe('BreadCrumbs', () => {
  const defaultProps = {
    state: {
      name: 'New York',
      slug: 'new-york',
    },
    city: {
      name: 'New York',
      slug: 'new-york',
    },
    headingText: 'Aliens have arrived',
    singlePropertyType: true,
  }

  it('is collapsed by default', () => {
    const wrapper = mount(<BreadCrumbs {...defaultProps} />)
    const expected = true
    const actual = wrapper.state().collapsed
    expect(actual).toBe(expected)
  })

  it('requires propertyType to render', () => {
    const wrapper = mount(<BreadCrumbs />)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('is un-collapsed when expand-crumbs btn clicked', () => {
    const wrapper = mount(<BreadCrumbs {...defaultProps} />)
    const crumb = wrapper.find('[data-tid="expand-crumbs"]')
    crumb.simulate('click')
    const expected = false
    const actual = wrapper.state().collapsed
    expect(actual).toEqual(expected)
  })

  describe('state+city, college, or military search', () => {
    const props = {
      ...defaultProps,
      hoodLanderUrl: defaultProps.city.slug,
    }
    it('renders collapsed', () => {
      const wrapper = shallow(<BreadCrumbs {...props} />)
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('renders exapanded', () => {
      const wrapper = shallow(<BreadCrumbs
        {...props}
        collapsed={false}
      />)
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('zip search', () => {
    const props = {
      ...defaultProps,
      zip: '12345',
      hoodLanderUrl: defaultProps.city.slug,
    }

    it('renders collapsed', () => {
      const wrapper = shallow(<BreadCrumbs
        {...props}
      />)
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('renders exapanded', () => {
      const wrapper = shallow(<BreadCrumbs
        {...props}
        collapsed={false}
      />)
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('hood search', () => {
    const props = {
      ...defaultProps,
      hood: {
        name: 'Greenwich Village',
        slug: 'greenwich-village',
      },
      hoodLanderUrl: defaultProps.city.slug,
    }

    it('renders collapsed', () => {
      const wrapper = shallow(<BreadCrumbs
        {...props}
      />)
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('renders exapanded', () => {
      const wrapper = shallow(<BreadCrumbs
        {...props}
        collapsed={false}
      />)
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
