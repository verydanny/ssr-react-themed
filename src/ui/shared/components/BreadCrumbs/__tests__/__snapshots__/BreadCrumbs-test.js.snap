// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumbs hood search renders collapsed 1`] = `

<div id="results"
     data-tid="breadcrumbs"
>
  <ol itemprop="breadcrumb"
      itemscope
      itemtype="https://schema.org/BreadcrumbList"
      data-tid="expandedBreadCrumbsText"
  >
    <li itemprop="itemListElement"
        itemscope
        itemtype="https://schema.org/ListItem"
    >
      <span itemprop="name">
        Home
      </span>
      <meta itemprop="position"
            content="1"
      >
    </li>
    <li>
      <a role="presentation"
         data-tid="expand-crumbs"
      >
        …
      </a>
    </li>
    <li class="undefined"
        itemprop="itemListElement"
        itemscope
        itemtype="https://schema.org/ListItem"
    >
      <meta itemprop="position"
            content="2"
      >
      <a href="/new-york"
         itemprop="url"
      >
        <span itemprop="name">
          New York
        </span>
      </a>
    </li>
    <li class="undefined"
        itemprop="itemListElement"
        itemscope
        itemtype="https://schema.org/ListItem"
    >
      <meta itemprop="position"
            content="3"
      >
      <a href="/new-york"
         itemprop="url"
      >
        <span itemprop="name">
          New York
        </span>
      </a>
    </li>
    <li class="undefined"
        itemprop="itemListElement"
        itemscope
        itemtype="https://schema.org/ListItem"
    >
      <meta itemprop="position"
            content="4"
      >
      <span itemprop="name">
        Greenwich Village
      </span>
    </li>
  </ol>
  <h1>
    Aliens have arrived
  </h1>
</div>

`;

exports[`BreadCrumbs hood search renders exapanded 1`] = `

<div id="results"
     data-tid="breadcrumbs"
>
  <ol itemprop="breadcrumb"
      itemscope
      itemtype="https://schema.org/BreadcrumbList"
      data-tid="expandedBreadCrumbsText"
  >
    <li itemprop="itemListElement"
        itemscope
        itemtype="https://schema.org/ListItem"
    >
      <span itemprop="name">
        Home
      </span>
      <meta itemprop="position"
            content="1"
      >
    </li>
    <li class
        itemprop="itemListElement"
        itemscope
        itemtype="https://schema.org/ListItem"
    >
      <meta itemprop="position"
            content="2"
      >
      <a href="/new-york"
         itemprop="url"
      >
        <span itemprop="name">
          New York
        </span>
      </a>
    </li>
    <li class
        itemprop="itemListElement"
        itemscope
        itemtype="https://schema.org/ListItem"
    >
      <meta itemprop="position"
            content="3"
      >
      <a href="/new-york"
         itemprop="url"
      >
        <span itemprop="name">
          New York
        </span>
      </a>
    </li>
    <li class
        itemprop="itemListElement"
        itemscope
        itemtype="https://schema.org/ListItem"
    >
      <meta itemprop="position"
            content="4"
      >
      <span itemprop="name">
        Greenwich Village
      </span>
    </li>
  </ol>
  <h1>
    Aliens have arrived
  </h1>
</div>

`;

exports[`BreadCrumbs requires propertyType to render 1`] = `null`;

exports[`BreadCrumbs state+city, college, or military search renders collapsed 1`] = `

<div id="results"
     data-tid="breadcrumbs"
>
  <ol itemprop="breadcrumb"
      itemscope
      itemtype="https://schema.org/BreadcrumbList"
      data-tid="expandedBreadCrumbsText"
  >
    <li itemprop="itemListElement"
        itemscope
        itemtype="https://schema.org/ListItem"
    >
      <span itemprop="name">
        Home
      </span>
      <meta itemprop="position"
            content="1"
      >
    </li>
    <li>
      <a role="presentation"
         data-tid="expand-crumbs"
      >
        …
      </a>
    </li>
    <li class="undefined"
        itemprop="itemListElement"
        itemscope
        itemtype="https://schema.org/ListItem"
    >
      <meta itemprop="position"
            content="2"
      >
      <a href="/new-york"
         itemprop="url"
      >
        <span itemprop="name">
          New York
        </span>
      </a>
    </li>
    <li class="undefined"
        itemprop="itemListElement"
        itemscope
        itemtype="https://schema.org/ListItem"
    >
      <meta itemprop="position"
            content="3"
      >
      <span itemprop="name">
        New York
      </span>
    </li>
  </ol>
  <h1>
    Aliens have arrived
  </h1>
</div>

`;

exports[`BreadCrumbs state+city, college, or military search renders exapanded 1`] = `

<div id="results"
     data-tid="breadcrumbs"
>
  <ol itemprop="breadcrumb"
      itemscope
      itemtype="https://schema.org/BreadcrumbList"
      data-tid="expandedBreadCrumbsText"
  >
    <li itemprop="itemListElement"
        itemscope
        itemtype="https://schema.org/ListItem"
    >
      <span itemprop="name">
        Home
      </span>
      <meta itemprop="position"
            content="1"
      >
    </li>
    <li class
        itemprop="itemListElement"
        itemscope
        itemtype="https://schema.org/ListItem"
    >
      <meta itemprop="position"
            content="2"
      >
      <a href="/new-york"
         itemprop="url"
      >
        <span itemprop="name">
          New York
        </span>
      </a>
    </li>
    <li class
        itemprop="itemListElement"
        itemscope
        itemtype="https://schema.org/ListItem"
    >
      <meta itemprop="position"
            content="3"
      >
      <span itemprop="name">
        New York
      </span>
    </li>
  </ol>
  <h1>
    Aliens have arrived
  </h1>
</div>

`;

exports[`BreadCrumbs zip search renders collapsed 1`] = `

<div id="results"
     data-tid="breadcrumbs"
>
  <ol itemprop="breadcrumb"
      itemscope
      itemtype="https://schema.org/BreadcrumbList"
      data-tid="expandedBreadCrumbsText"
  >
    <li itemprop="itemListElement"
        itemscope
        itemtype="https://schema.org/ListItem"
    >
      <span itemprop="name">
        Home
      </span>
      <meta itemprop="position"
            content="1"
      >
    </li>
    <li>
      <a role="presentation"
         data-tid="expand-crumbs"
      >
        …
      </a>
    </li>
    <li class="undefined"
        itemprop="itemListElement"
        itemscope
        itemtype="https://schema.org/ListItem"
    >
      <meta itemprop="position"
            content="2"
      >
      <a href="/new-york"
         itemprop="url"
      >
        <span itemprop="name">
          New York
        </span>
      </a>
    </li>
    <li class="undefined"
        itemprop="itemListElement"
        itemscope
        itemtype="https://schema.org/ListItem"
    >
      <meta itemprop="position"
            content="3"
      >
      <a href="/new-york"
         itemprop="url"
      >
        <span itemprop="name">
          New York
        </span>
      </a>
    </li>
    <li class="undefined"
        itemprop="itemListElement"
        itemscope
        itemtype="https://schema.org/ListItem"
    >
      <meta itemprop="position"
            content="4"
      >
      <span itemprop="name">
        12345
      </span>
    </li>
  </ol>
  <h1>
    Aliens have arrived
  </h1>
</div>

`;

exports[`BreadCrumbs zip search renders exapanded 1`] = `

<div id="results"
     data-tid="breadcrumbs"
>
  <ol itemprop="breadcrumb"
      itemscope
      itemtype="https://schema.org/BreadcrumbList"
      data-tid="expandedBreadCrumbsText"
  >
    <li itemprop="itemListElement"
        itemscope
        itemtype="https://schema.org/ListItem"
    >
      <span itemprop="name">
        Home
      </span>
      <meta itemprop="position"
            content="1"
      >
    </li>
    <li class
        itemprop="itemListElement"
        itemscope
        itemtype="https://schema.org/ListItem"
    >
      <meta itemprop="position"
            content="2"
      >
      <a href="/new-york"
         itemprop="url"
      >
        <span itemprop="name">
          New York
        </span>
      </a>
    </li>
    <li class
        itemprop="itemListElement"
        itemscope
        itemtype="https://schema.org/ListItem"
    >
      <meta itemprop="position"
            content="3"
      >
      <a href="/new-york"
         itemprop="url"
      >
        <span itemprop="name">
          New York
        </span>
      </a>
    </li>
    <li class
        itemprop="itemListElement"
        itemscope
        itemtype="https://schema.org/ListItem"
    >
      <meta itemprop="position"
            content="4"
      >
      <span itemprop="name">
        12345
      </span>
    </li>
  </ol>
  <h1>
    Aliens have arrived
  </h1>
</div>

`;
